test/common/snap.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+const common = require('./');
+const path = require('node:path');
+const fs = require('node:fs/promises');
+const assert = require('node:assert/strict');
+
+
+const stackFramesRegexp = /(\s+)((.+?)\s+\()?(?:\(?(.+?):(\d+)(?::(\d+))?)\)?(\s+\{)?(\n|$)/g;
+
+function replaceStackTrace(str) {
+  return str.replace(stackFramesRegexp, '$1*$7\n');
+}
+
+function transform(...args) {
+  return (str) => args.reduce((acc, fn) => fn(acc), str);
+}
+
+function getSnapshotPath(filename) {
+  const { name, dir } = path.parse(filename);
+  return path.resolve(dir, `${name}.snapshot`);
+}
+
+async function assertSnapshot(actual, filename = process.argv[1]) {
+  const snapshot = getSnapshotPath(filename);
+  const expected = process.env.NODE_REGENERATE_SNAPSHOTS ? actual : await fs.readFile(snapshot, 'utf8');
+  if (process.env.NODE_REGENERATE_SNAPSHOTS) {
+    await fs.writeFile(snapshot, actual);
+  }
+  assert.strictEqual(actual, expected);
```

Comment:
No need to define `expected` and then `assert.strictEqual(actual, expected)` if you are replacing the file anyway.

```suggestion
  if (process.env.NODE_REGENERATE_SNAPSHOTS) {
    await fs.writeFile(snapshot, actual);
  } else {
    const expected = await fs.readFile(snapshot, 'utf8');
    assert.strictEqual(actual, expected);
  }
```